src/block_cache/block_chunkcache.c

```diff
@@ -845,6 +846,43 @@ __wt_chunkcache_remove(
     return (0);
 }
 
+/*
+ * __wt_chunkcache_unpin_old_versions --
+ *     Unpin the old versions of newly added chunks so eviction can remove them.
+ */
+void
+__wt_chunkcache_unpin_old_versions(WT_SESSION_IMPL *session, const char *sp_obj_name)
+{
+    WT_CHUNKCACHE *chunkcache;
+    WT_CHUNKCACHE_CHUNK *chunk, *chunk_tmp;
+    int i;
+
+    chunkcache = &S2C(session)->chunkcache;
+
+    /* Optimization: check if the file contains objects in the pinned list, otherwise skip. */
+    if (__chunkcache_should_pin_chunk(session, sp_obj_name)) {
+        /* Loop through the entire chunkcache and search for matching objects from the file and
+         * clear the pinned flag. */
+        for (i = 0; i < (int)chunkcache->hashtable_size; i++) {
+            __wt_spin_lock(session, &chunkcache->hashtable[i].bucket_lock);
+            TAILQ_FOREACH_SAFE(chunk, WT_BUCKET_CHUNKS(chunkcache, i), next_chunk, chunk_tmp)
+            {
+                if (strcmp(chunk->hash_id.objectname, sp_obj_name) == 0) {
+                    if (F_ISSET(chunk, WT_CHUNK_PINNED)) {
+                        /* Decrement the stat when a chunk that was initially pinned becomes
```

Comment:
Nitpick: I realise `s_all` probably formatted it this way, but what about formatting like:
```
/*
 * Decrement the stat ...
 * unpinned.
 */
```

It looks kinda weird at the moment.

Reply:
done here [de67e9a](https://github.com/wiredtiger/wiredtiger/pull/9730/commits/de67e9a805f3926a1d3384e7dd1c81db5411c3db)

Reply:
I can see `s_all` put them back - I disagree with `s_all` but it's a shell script so I can't win this argument ;-)